internal/junitxml/report.go

```diff
@@ -224,11 +232,26 @@ func packageTestCases(pkg *testjson.Package, formatClassname FormatFunc) []JUnit
 }
 
 func newJUnitTestCase(tc testjson.TestCase, formatClassname FormatFunc) JUnitTestCase {
-	return JUnitTestCase{
+	jtc := JUnitTestCase{
 		Classname: formatClassname(tc.Package),
 		Name:      tc.Test.Name(),
 		Time:      formatDurationAsSeconds(tc.Elapsed),
 	}
+
+	if len(tc.Attributes) > 0 {
+		var properties []JUnitProperty
+		for k, v := range tc.Attributes {
+			p := JUnitProperty{Name: k, Value: v}
+			properties = append(properties, p)
+		}
+		sort.Slice(properties, func(i, j int) bool {
+			return properties[i].Name < properties[j].Name
+		})
```

Comment:
sort.Slice is somehow discouraged.

https://pkg.go.dev/sort#Sort
> Note: in many situations, the newer slices.SortFunc function is more ergonomic and runs faster. 

[slices.SortFunc](https://pkg.go.dev/slices#SortFunc) is a good option.

Also in the function you can use cmp.Compare that can be used directly

https://pkg.go.dev/cmp#Compare

So it ends up to something like this (pseudo code written on a phone)

```suggestion
		slices.Sort(properties, func(a, b JUnitProperty) int {
			return cmp.Compare(a.Name, b.Name)
		})
```